bottlecap/src/appsec/processor.rs

```diff
@@ -0,0 +1,1361 @@
+use std::collections::HashMap;
+use std::sync::{Arc, Mutex};
+use std::time::{Duration, Instant};
+
+use crate::appsec::sampler::Sampler;
+use crate::appsec::{is_enabled, is_standalone, payload};
+use crate::config::Config;
+
+use bytes::Bytes;
+use libddwaf::object::{WafMap, WafObject, WafOwned};
+use libddwaf::{waf_map, Builder, Config as WAFConfig, Context, Handle, RunResult};
+use tracing::{debug, info, warn};
+/// The App & API Protection processor.
+///
+/// It is used to try to identify invoke requests that are supported, extract the relevant data from
+/// the request payload, and evaluate in-app WAF rules against that data.
+pub struct Processor {
+    handle: Handle,
+    waf_timeout: Duration,
+    api_sec_sampler: Mutex<Sampler>,
+}
+impl Processor {
+    /// Creates a new [`Processor`] instance using the provided [`Config`].
+    ///
+    /// # Errors
+    /// - If [`Config::serverless_appsec_enabled`] is `false`;
+    /// - If the [`Config::appsec_rules`] points to a non-existent file;
+    /// - If the [`Config::appsec_rules`] points to a file that is not a valid JSON-encoded ruleset;
+    /// - If the in-app WAF fails to initialize, integrate the ruleset, or build the WAF instance.
+    pub fn new(config: &Config) -> Result<Self, Error> {
+        if !is_enabled(config) {
+            return Err(Error::FeatureDisabled);
+        }
+        debug!("Starting ASM processor");
+
+        if is_standalone() {
+            info!(
```

Comment:
Should this be a warn?

Reply:
In principle this is the result of an intentional action by the user... so I doubt it's relevant to _warn_ here? But I don't feel all too strongly about that.